### PR TITLE
Contributors can add the ready tag to their own prs

### PR DESCRIPTION
**Pull request by JJRcop**
I can't test this so I don't know if it works.

Assimilates, and thus closes #32052

### CODE REVIEW

**Comment by ChangelingRain** [comment]
Now that the label has been renamed, this PR is no longer relevant.

**Comment by JJRcop** [comment]
This PR allows contributors to tag their own prs as ready with the Needs Review tag, and it actually does the proper renaming of the label to 'Needs Review' in the php script.

**Comment by JJRcop** [comment]
@optimumtact

**Comment by optimumtact** [comment]
I'm interested in having this change.

It's easier for me to search by label and only see ones that are nominally ready for actual review

File: tools/WebhookProcessor/github_webhook_processor.php [hunk 1]
```diff
@@ -217,8 +217,8 @@ function tag_pr($payload, $opened) {
 			$tags[] = $tag;
 
 	check_tag_and_replace($payload, '[dnm]', 'Do Not Merge', $tags);
-	if(!check_tag_and_replace($payload, '[wip]', 'Work In Progress', $tags))
-		check_tag_and_replace($payload, '[ready]', 'Work In Progress', $remove);
+	if(!check_tag_and_replace($payload, '[wip]', 'Work In Progress', $tags) && check_tag_and_replace($payload, '[ready]', 'Work In Progress', $remove))
+		$tags[] = 'Needs Review';
```
Comment (Cyberboss) [inline tools/WebhookProcessor/github_webhook_processor.php]: This shouldn't be able to bypass PRs with outstanding changes requested reviews

**Review by Cyberboss — DISMISSED** [review]

File: tools/WebhookProcessor/github_webhook_processor.php (hunk 1 above)
Comment (JJRcop) [reply tools/WebhookProcessor/github_webhook_processor.php]: The way I understand it, this is run before the check_ready_for_review function, so the label will be removed with outstanding changes requested reviews.

https://github.com/tgstation/tgstation/blob/317a93cc1bd2f3d59fdbd183cc3c0653bb1128d2/tools/WebhookProcessor/github_webhook_processor.php#L377-L379

Comment (Cyberboss) [reply tools/WebhookProcessor/github_webhook_processor.php]: Labels are still applied at the end of this function though, so you'll end up with events like `tgstation-server added 'Needs Review' and removed 'Needs Review' 10 minutes ago`

Comment (JJRcop) [reply tools/WebhookProcessor/github_webhook_processor.php]: So then the solution is to remove the apisend from both functions, and instead have them return the labels they want to add, and the labels they want to remove, then add this block to the end of the synchronize case. (Or put it in its own function)
https://github.com/JJRcop/tgstation/blob/317a93cc1bd2f3d59fdbd183cc3c0653bb1128d2/tools/WebhookProcessor/github_webhook_processor.php#L225-L236

im a little busy so i'll do this later

Comment (JJRcop) [reply tools/WebhookProcessor/github_webhook_processor.php]: https://github.com/tgstation/tgstation/pull/32062/commits/3f9e1f6cf108bcbab05ed60c1207fb4cd254fba1

Comment (JJRcop) [reply tools/WebhookProcessor/github_webhook_processor.php]: This issue was addressed.

**Comment by JJRcop** [comment]
@Cyberboss I addressed your issue, could you give feedback, or dismiss it?

**Comment by optimumtact** [comment]
he never does

**Comment by JJRcop** [comment]
#32297 is blocking

**Pull request merged**